analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantNullCheck.cs

```diff
@@ -165,7 +165,7 @@ private void CheckOrPattern(SyntaxNodeAnalysisContext context)
             // Verifies if the given pattern syntax is like "not X" - where X can be constant pattern (except 'null'), Declaration pattern, Recursive pattern.
             // The PatternSyntax appears e.g. in switch arms and is different from IsPatternSyntax.
             static bool IsNegativePatternMatch(PatternSyntaxWrapper patternSyntaxWrapper) =>
-                patternSyntaxWrapper.IsNot() && !patternSyntaxWrapper.IsNull();
```

Comment:
I'm pretty sure I am missing something in the test cases, but they are green and the code coverage was incomplete, thus that previous logic was not necessary

Reply:
You are right. Actually, IsNull requires ConstantPattern and IsNot NotPattern, so even  `not null` in `null or not null` won't match it.

Reply:
While investigating it, I've found a FP we should be aware of 
```
x = a switch
{
    null or not null => ""
};
```
Even the check makes no sense, removing null from it is a bad idea, since then it won't match null cases. WDYT?

Reply:
Thank you, this was it! I was a bit tired with all the test cases. Will fix it.